sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -223,6 +224,64 @@ public Node visitBegin(SqlBaseParser.BeginContext context) {
         return new BeginStatement();
     }
 
+    @Override
+    public Node visitIntervalLiteral(SqlBaseParser.IntervalLiteralContext context) {
+        IntervalLiteral.IntervalField startField = getIntervalFieldType((Token) context.from.getChild(0).getPayload());
+        IntervalLiteral.IntervalField endField = null;
+        if (context.to != null) {
+            Token token = (Token) context.to.getChild(0).getPayload();
+            endField = getIntervalFieldType(token);
+        }
+
+        if (endField != null) {
+            if (startField.compareTo(endField) > 0) {
+                throw new IllegalArgumentException(
+                    String.format("Startfield %s must be less significant than Endfield %s", startField, endField));
+            }
+        }
+
+        IntervalLiteral.Sign sign = IntervalLiteral.Sign.PLUS;
+        if (context.sign != null) {
+            sign = getIntervalSign(context.sign);
+        }
+
+        return new IntervalLiteral(
+            ((StringLiteral) visit(context.stringLiteral())).getValue(),
+            sign,
+            startField,
+            endField);
+    }
+
+    private static IntervalLiteral.Sign getIntervalSign(Token token) {
+        switch (token.getType()) {
+            case SqlBaseLexer.MINUS:
+                return IntervalLiteral.Sign.MINUS;
+            case SqlBaseLexer.PLUS:
+                return IntervalLiteral.Sign.PLUS;
+            default:
+                throw new IllegalArgumentException("Unsupported sign: " + token.getText());
+        }
+    }
+
+    private static IntervalLiteral.IntervalField getIntervalFieldType(Token token) {
```

Comment:
Given that the `IntervalField` is never used but directly converted to a `IntervalParser.Precision` at the expression analyzer, maybe it makes sense to use `IntervalParser.Precision.` already here and get rid of the `IntervalField`?

Reply:
Yes, good idea.

Reply:
This would introduce a new dependency between sql-parser and common which is currently not there. If this ok with you, i will do it.

Reply:
Hm right, so ok lets leave it like that.